### ISSUE TITLE
Review documents endpoint segmentation

### ISSUE THREAD

**Issue by AshDeeming** [reapit/foundations#7929]
Following feedback (see ZenDesk ticket #380) 

Review access to certain endpoints where sensitive data may be stored

**Comment by plittlewood-rpt** [comment]
PL/HP reviewed - need new IdP and there are operational complexities with doing this

**Comment by plittlewood-rpt** [comment]
@HollyJoyPhillips I don't think we should do this until we can filter by the private flag as we are already seeing messages from developers about how the office filtering has had to be implemented and this is potentially even more confusing. I've raise the issue of not being able to filter performantly by this field with the desktop team to see if we should move on the alternative data store piece of work or wait until they've structured the field

**Comment by plittlewood-rpt** [comment]
Internal ref: PBI-4664
